astropy/coordinates/solar_system.py

```diff
@@ -56,116 +68,91 @@ def _download_spk_file(url=('http://naif.jpl.nasa.gov/pub/naif/'
 
     .. [1] http://naif.jpl.nasa.gov/pub/naif/generic_kernels/spk/planets/aareadme_de430-de431.txt
     """
+    de_number = int(ephemeris[2:])
+    if not ephemeris.startswith('de') or not (0 < de_number < 1000):
+        raise ValueError("JPL ephemeris names be 'deNNN'.")
+    url = ('http://naif.jpl.nasa.gov/pub/naif/'
+           'generic_kernels/spk/planets/de{0:03d}.bsp'.format(de_number))
+
     return download_file(url, cache=True, show_progress=show_progress)
 
 
-def _get_kernel(*args, **kwargs):
+def _get_kernel(ephemeris='de430'):
     """
     Try importing jplephem, download/retrieve from cache the Satellite Planet
-    Kernel.
+    Kernel corresponding to the given ephemeris.
     """
     global KERNEL
 
-    try:
-        from jplephem.spk import SPK
-    except ImportError:
-        raise ImportError("Solar system ephemeris calculations require the "
-                          "jplephem package "
-                          "(https://pypi.python.org/pypi/jplephem)")
+    ephemeris = ephemeris.lower()
+
+    if KERNEL is None or KERNEL.name != ephemeris:
+        try:
+            from jplephem.spk import SPK
+        except ImportError:
+            raise ImportError("Solar system JPL ephemeris calculations require "
+                              "the jplephem package "
+                              "(https://pypi.python.org/pypi/jplephem)")
 
-    if KERNEL is None:
-        KERNEL = SPK.open(_download_spk_file())
+        KERNEL = SPK.open(_download_spk_file(ephemeris))
+        KERNEL.name = ephemeris
     return KERNEL
 
 
-def get_body_barycentric(time, body):
+def get_body_barycentric(body, time, ephemeris=None):
```

Comment:
From an API point of view, I'm happy to swap body and time, and also to take the location from the time if none is supplied. 

I'm not sure about having the ephemeris as the argument to the functions. If a user makes a series of calls and forgets to set the ephemeris in one, the behaviour might be quite unpredictable. One ought, IMO, to be able to set the state at the start of your code and the same ephemeris will be used throughout.